src/ClientCommand.cpp

```diff
@@ -726,7 +726,8 @@ void CClient::UserCommand(CString& sLine) {
             return;
         }
 
-        if (pNewUser->FindNetwork(sNewNetwork)) {
+        if (pNewUser->FindNetwork(sNewNetwork) &&
+            (!sOldUser.Equals(sNewUser) || !sOldNetwork.Equals(sNewNetwork))) {
```

Comment:
Why are you comparing username?

Reply:
It appears two users can have networks with equal names, and in the "movenetwork A znc B znc" case, the error should be given even though the network names are equal.

Reply:
The username is already converted to the pointer to CUser. In a case-sensitive way. Compare pointers instead of reimplementing the existing logic (but with subtly different semantics)